internal/querynode/query_collection.go

```diff
@@ -95,1315 +98,123 @@ func newQueryCollection(releaseCtx context.Context,
-					colData = append(colData, data)
-				}
-			}
-			newCol := &schemapb.FieldData{
-				Field: &schemapb.FieldData_Scalars{
-					Scalars: &schemapb.ScalarField{
-						Data: &schemapb.ScalarField_DoubleData{
-							DoubleData: &schemapb.DoubleArray{
-								Data: colData,
-							},
-						},
-					},
-				},
-			}
-			finalResult.FieldsData = append(finalResult.FieldsData, newCol)
-			blobOffset += blobLen
-		case schemapb.DataType_FloatVector:
-			dim, err := schema.GetVectorDimFromID(fieldID)
-			if err != nil {
-				return nil, err
-			}
-			blobLen := dim * 4
-			var colData []float32
-			for _, hit := range hits {
-				for _, row := range hit.RowData {
-					dataBlob := row[blobOffset : blobOffset+blobLen]
-					//ref https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices
-					ptr := unsafe.Pointer(&dataBlob[0])
-					farray := (*[1 << 28]float32)(ptr)
-					colData = append(colData, farray[:dim:dim]...)
-				}
-			}
-			newCol := &schemapb.FieldData{
-				Field: &schemapb.FieldData_Vectors{
-					Vectors: &schemapb.VectorField{
-						Dim: int64(dim),
-						Data: &schemapb.VectorField_FloatVector{
-							FloatVector: &schemapb.FloatArray{
-								Data: colData,
-							},
-						},
-					},
-				},
-			}
-			finalResult.FieldsData = append(finalResult.FieldsData, newCol)
-			blobOffset += blobLen
-		case schemapb.DataType_BinaryVector:
-			dim, err := schema.GetVectorDimFromID(fieldID)
-			if err != nil {
-				return nil, err
-			}
-			blobLen := dim / 8
-			var colData []byte
-			for _, hit := range hits {
-				for _, row := range hit.RowData {
-					dataBlob := row[blobOffset : blobOffset+blobLen]
-					colData = append(colData, dataBlob...)
-				}
-			}
-			newCol := &schemapb.FieldData{
-				Field: &schemapb.FieldData_Vectors{
-					Vectors: &schemapb.VectorField{
-						Dim: int64(dim),
-						Data: &schemapb.VectorField_BinaryVector{
-							BinaryVector: colData,
-						},
-					},
-				},
-			}
-			finalResult.FieldsData = append(finalResult.FieldsData, newCol)
-			blobOffset += blobLen
-		default:
-			return nil, fmt.Errorf("unsupported data type %s", schemapb.DataType_name[int32(fieldMeta.DataType)])
-		}
-	}
-
-	return finalResult, nil
-}
-
-// TODO:: cache map[dsl]plan
-// TODO: reBatched search requests
-func (q *queryCollection) search(msg queryMsg) error {
-	searchMsg := msg.(*msgstream.SearchMsg)
-	sp, ctx := trace.StartSpanFromContext(searchMsg.TraceCtx())
-	defer sp.Finish()
-	searchMsg.SetTraceCtx(ctx)
-	searchTimestamp := searchMsg.BeginTs()
-	travelTimestamp := searchMsg.TravelTimestamp
-
-	collectionID := searchMsg.CollectionID
-	collection, err := q.streaming.replica.getCollectionByID(collectionID)
-	if err != nil {
-		return err
-	}
-	schema, err := typeutil.CreateSchemaHelper(collection.schema)
+	// create query stages
+	col, err := q.streaming.replica.getCollectionByID(q.collectionID)
 	if err != nil {
 		return err
 	}
+	channels := col.getVChannels()
```

Comment:
what if querynode receieve a watchDmChannel request for exist collection?

Reply:
would be fixed in next feature